Add plugin: aDHL

tine-schreibt opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin:

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [ ]  Linux **(don't have access to a Linux machine but I see no reason why it wouldn't work there)**
  - [x]  Android _(if applicable)_ 
  - [ ]  iOS _(if applicable)_ **(same here; no access but also no reason why it wouldn't work)**
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

Please, dear bot, do another validation check. I've done every fucking thing I could think of to make you spring into action, and this is the last.

joethei commented:
Sorry for the delay in reviewing.

[Copyright (c) 2024 CS](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/LICENSE.txt#L3), ["author": "tine-schreibt",](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/manifest.json#L7)
Please also add credits to the original author(s)

["description": "The Dynamic Highlights Plugin, but with hotkeys, more options and sorting; works well with Highlightr.",](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/manifest.json#L6)
Please don't compare to other plugins here, (yes, that includes the original)
We have published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.

[name: `The Switch - Start/stop all static highlighting`,](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/main.ts#L116), [text: "Persistent Highlights",](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/ui.ts#L182), [saveButton.buttonEl.setAttribute("aria-label", "Save Highlighter");](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/ui.ts#L453), [.setTooltip("Discard Changes")](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/ui.ts#L704), [chooseCommands.setName("Hotkeys and Command Palette");](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/ui.ts#L1097), [text: "Selection Highlights",](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/ui.ts#L1121), [.setTooltip("Save CSS Snippet")](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/ui.ts#L1227), [contentEl.createEl("h2", { text: "Create New tag" });](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/modals.ts#L36), [.setTooltip("Save CSS Snippet")](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/ui.ts#L1227)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[export interface Workspace {](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/types/obsidian.d.ts#L4-L6)
This is part of the public API, no need to declare it here.

[console.log("accentColor: ", hslaColor);](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/ui.ts#L169), [console.log("New queryOrder after unshift:", config.queryOrder);](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/ui.ts#L518),, [console.log("Copied to clipboard!");](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/export.ts#L49)
Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.

[headlineToggleContainer.createEl("h3", {](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/ui.ts#L181-L184), [containerEl.createEl("h3", {](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/ui.ts#L1120-L1123)
For section headings in settings use:
```ts
new Setting(containerEl).setName('name here').setHeading();
```

[saveButton.buttonEl.setAttribute("aria-label", "Save Highlighter");](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/ui.ts#L453), [cssSaveButton.buttonEl.setAttribute("aria-label", "Save Highlighter");](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/ui.ts#L1221), [selectedDiscardButton.buttonEl.setAttribute(](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/ui.ts#L1246-L1249)
Remove this code, it's overwritten by the call to `setTooltip` below.

[.setClass("mod-cta")](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/ui.ts#L702)
Use the `setCta` function instead.

[saveButton.setAttribute("aria-label", "Save new tag.");](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/modals.ts#L50), [saveButton.setAttribute("aria-label", "Save new tag name.");](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/modals.ts#L106), [deleteButton.setAttribute("aria-label", `Delete ${this.oldTagName}.`);](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/modals.ts#L191), [deleteButton.setAttribute("aria-label", `Delete ${this.oldTagName}.`);](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/modals.ts#L191), [deleteButton.setAttribute("aria-label", `Delete ${this.highlighterName}.`);](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/modals.ts#L271), [cancelButton.setAttribute("aria-label", "Cancel.");](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/settings/modals.ts#L292)
You should use the `setTooltip` function for these.

["save":](https://github.com/tine-schreibt/aDHL/blob/64eb271dae6a27f6ae3b86eacc2c3c76f1b1fc04/src/icons/customIcons.ts#L4)
There is a icon with the name `save` in lucide already, either remove this code if that's the icon you want to use, or rename this.

tine-schreibt commented:
Hi. 
No problem. Hope you're doing okay :) 

Question: 
The reference https://docs.obsidian.md/Reference/Manifest specifies no property for additional authors or contributors, which is why I only mention/link them in my Readme. I completely forgot to re-read the MIT license, though -.- 
So, do I just put "tine schreibt; NothingIsLost" as value for the "author" property in the manifest? Also, do I copy the original author's URL over as well? 

Also: I'm so sorry for getting mixed up on which guide I'd actually read -.- I could have saved you a lot of work. 
There's nothing about comparisons in the style guide, though (unless I missed something), so an update there would be helpful. 

I hope you have a relatively stress free weekend.

tine-schreibt commented:
Okay.
I changed everything except for the attribution in manifest.js, since I don't yet know how. 
I...
- added the attribution from the original plugin's license to my license.
- changed the description in manifest.js to no longer compare.
- changed everything in the UI to sentence case (I think...).
- deleted the relics that were 'interface Workspace' and icons.js.
- deleted all the console.logs() that I forgot.
- refactored all section headings.
- replaced aria-labels with tooltips, except in a couple of instances where it's not possible; there I used the "has-tooltip" class (I'm not sure if that's really the correct way to do it, so let me know if I need to change these instances, too, and how I should change them)
- changed all instances of .setClass("mod-cta") to .setCta() (and all instances of .setClass("mod-warn") to .setWarning()) (question here: I have some instances where I use the inverted button style; I couldn't find a way to do that without using setClass(). If there is one, please let me know.)

I also cleaned up the UI a bit more and reworded/refactored some stuff to make it easier to understand. 

I hope I didn't miss anything/didn't accidentally introduce new errors. 
Thanks for you patience with my n00b self :)

joethei commented:
One thing left, the license of the Highlightr plugin requires that every thing that uses code from that plugin is licensed under the Mozilla Public License 2.0 or a compatible license (MIT is not compatible).
You will need to change the license of your plugin if you want to keep using that code.

tine-schreibt commented:
Ah, good catch. 
To avoid having to hybrid or double licence or talk to people, I went and recreated the css styles in my own way so they are now original code by me and can be licensed however I want. 
I didn't change the attribution in manifest.js, though, since you didn't give me any direction for it. Is that off the table? 
Have a better week than expected! :)

tine-schreibt commented:
So...............?

tine-schreibt commented:
I'm waiting for the reviewer to release it!

tine-schreibt commented:
Ugh, sorry. I fixed it, please try again -.-

tine-schreibt commented:
God, this sucks.

tine-schreibt commented:
Both manifest.json point to version 1.0.4 now, which is the same that I tagged the release with. So please. Please...........

tine-schreibt commented:
Dear reviewer, can you please explain to me what I am doing wrong? Both my manifest.json - in my repo and in the release -  now point to version 1.0.4, which is also the version that the release is tagged with. 

What does the bot want from me? What am I missing? Please respond. I'm feeling abandoned -.-

tine-schreibt commented:
I've set the release tag back to 1.0.1, same for the version number in manifest.json. If it still doesn't work I'm going to scream.

tine-schreibt commented:
Error when fetching the repository
  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin +refs/pull/5150/merge:refs/remotes/pull/5150/merge
  Error: fatal: couldn't find remote ref refs/pull/51[50](https://github.com/obsidianmd/obsidian-releases/actions/runs/14695119523/job/41235682992?pr=5150#step:2:55)/merge
  The process '/usr/bin/git' failed with exit code 128
 
Please help, thank you.

whitenoisedev commented:
You manifest.json says 1.0.1, the only **release** that you have is 1.0.3.

tine-schreibt commented:
Look at that. There's twenty *five* places where one needs to manually change the version number, and not twenty *four* like I had assumed.
I set everything I know about to version 1.0.3 now, even thought the error I'm getting doesn't seem to refer to versioning. We'll see if this fixed anything.

tine-schreibt commented:
Nope.
  Error: fatal: couldn't find remote ref refs/pull/5150/merge
  Error: The process '/usr/bin/git' failed with exit code 128

joethei commented:
This is a pretty strange error, we don't understand why it happens and how to fix it.
The workaround is to rebase and then force push
